src/infra/other-installation-methods.md

```diff
@@ -104,8 +104,62 @@ Past releases can be found in [the archives].
 
 ## Source code
 
+If you want to build the Rust toolchain from source code, you can use the following
+links to download source code tarballs.
+
 {{#source_code_table}}
 
+If you want to make sure that the published source tarball matches what is in the
+`rust` git repository, you can use the following script as a template:
+
+<details>
+<summary>Script for reproducing source tarball contents</summary>
+
+```bash
+#!/bin/bash
+
+set -e
+
+# You can use either a commit SHA or a stable release version (e.g. 1.XY.Z)
+TAG=a8cfc83801301c2b4f0fd030192e268eeb15d473
+# TAG=1.77.1
+
+# Clone Rust toolchain repositor from GitHub
+git clone https://github.com/rust-lang/rust
+cd rust
+git reset --hard ${TAG}
```

Comment:
if it's just for building the tarball and the user doesn't care about git history, maybe can we suggest a faster clone?

```suggestion
# Clone Rust toolchain repository from GitHub
git clone --depth=1 https://github.com/rust-lang/rust
cd rust
git reset --hard ${TAG}
```

Reply:
We actually do care about the history, because you want to checkout a specific git tag in the repository :) `git reset` wouldn't work with `--depth=1`.
We also cannot use `--branch` and `--single-branch` because of detached HEAD, see https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/Separate.20CI.20workflow.20for.20source.20tarballs/near/430495074).

Reply:
Fixed the typo, thanks!